src/coreclr/clrdefinitions.cmake

```diff
@@ -53,6 +53,7 @@ if(CLR_CMAKE_HOST_WIN32)
   add_definitions(-D_WIN32_WINNT=0x0602)
   add_definitions(-DWIN32_LEAN_AND_MEAN)
   add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+  add_compile_definitions(NOMINMAX)
```

Comment:
Why do we use `add_compile_definitions` here and `add_definitions` above?

Reply:
I generally want to move us to `add_compile_definitions` in general as it's a little clearer (don't need the `-D` parameter) and propagates well to our custom functions logic like `get_compile_definitions`. I didn't want to bloat this PR further by changing all of the other ones in this PR.

Reply:
Ok, it makes sense then.